packages/typespec-python/src/emitter.ts

```diff
@@ -202,6 +202,12 @@ function getEffectiveSchemaType(context: DpgContext, type: Model): Model {
 
     const effective = getEffectiveModelType(program, type, isSchemaProperty);
     if (effective.name) {
+        if (effective.name === "ResourceOperationStatus") {
+            const result = effective.properties.get("result");
+            if (result?.type.kind === "Model") {
+                return result.type;
+            }
+        }
```

Comment:
This is just workaround. After https://github.com/Azure/typespec-azure/issues/2675 fixed, we can use `getLroMetadata` to resolve it.